shelf/hook/manager.py

```diff
@@ -0,0 +1,55 @@
+from shelf.hook.event import Event
+from shelf.hook.background import action
+
+
+class Manager(object):
+    def __init__(self, container, multiprocessing, host, command):
+        """
+            Args:
+                container(shelf.hook.container)
+                multiprocessing(multiprocessing)
+                host(string) The protocol + hostname which hosts shelf.
+                command(string) Command to run for each event.
+        """
+        self.container = container
+        self.multiprocessing = multiprocessing
+        self.host = host
+        self.command = command
+
+    def notify_artifact_uploaded(self, identity):
+        """
+            Should be called when an artifact is uploaded.
```

Comment:
Yeah I had the same thought. I don't think the function name is self explanatory but I don't know what its going to do, because that is up to the hook functionality.

I guess what I can say is that it will fire an event for an artifact being uploaded.